core/src/main/java/org/elasticsearch/index/shard/IndexShard.java

```diff
@@ -503,6 +536,10 @@ public void relocated(String reason) throws IllegalIndexShardStateException, Int
                         throw new IllegalIndexShardStateException(shardId, IndexShardState.STARTED,
                             ": shard is no longer relocating " + shardRouting);
                     }
+                    if (primaryReplicaResyncInProgress) {
```

Comment:
I'm not sure I understand the question. This will trigger the recovery process of the relocation to retry (and hopefully by then the resync will be completed).